x-pack/plugins/uptime/public/components/overview/alerts/uptime_alerts_flyout_wrapper.tsx

```diff
@@ -17,14 +17,15 @@ export const UptimeAlertsFlyoutWrapperComponent = ({
   alertFlyoutVisible,
   alertTypeId,
   setAlertFlyoutVisibility,
-}: Props) => (
-  <AlertAdd
-    addFlyoutVisible={alertFlyoutVisible}
-    consumer="uptime"
-    setAddFlyoutVisibility={setAlertFlyoutVisibility}
-    alertTypeId={alertTypeId}
-    // if we don't have an alert type pre-specified, we need to
-    // let the user choose
-    canChangeTrigger={!alertTypeId}
-  />
-);
+}: Props) =>
+  alertFlyoutVisible && alertTypeId ? (
+    <AlertAdd
+      addFlyoutVisible={alertFlyoutVisible}
+      consumer="uptime"
+      setAddFlyoutVisibility={setAlertFlyoutVisibility}
+      alertTypeId={alertTypeId}
+      // if we don't have an alert type pre-specified, we need to
+      // let the user choose
+      canChangeTrigger={!alertTypeId}
+    />
```

Comment:
Seems like Flyout will not open if alertTypeId isn't selected? but i guess that use case will never happen inside uptime, right?

Reply:
Yeah, exactly - we need to fix some buggy race conditions on our end which allow Uptime to accidentally render half a flyout that *thinks* no type is selected (because `alertTypeId` is `undefined`) and half that thinks you have selected one.

Uptime is the only place (that I'm aware of) where this behaviour is relied on... so the idea is to patch it this way [until we can address it properly](https://github.com/elastic/kibana/issues/83417) .